syntax = "proto3";
package redis_request;

enum RequestType {
    /// Invalid request type
    InvalidRequest = 0;
    /// An unknown command, where all arguments are defined by the user.
    CustomCommand = 1;
    /// Type of a get string request.
    GetString = 2;
    /// Type of a set string request.
    SetString = 3;
}

message RedisRequest {
    message ArgsArray {
        repeated string args = 1;
    }
    
    uint32 callback_idx = 1;
    RequestType request_type = 2;
    oneof args {
        ArgsArray args_array = 3;
        uint64 args_vec_pointer = 4;
    }
}
